src/frontend/src/handler/create_source.rs

```diff
@@ -118,7 +119,13 @@ pub async fn handle_create_source(
 ) -> Result<PgResponse> {
     let (column_descs, pk_column_id_from_columns) = bind_sql_columns(stmt.columns)?;
     let (mut columns, pk_column_ids, row_id_index) =
-        bind_sql_table_constraints(column_descs, pk_column_id_from_columns, stmt.constraints)?;
+        bind_sql_table_constraints(column_descs, pk_column_id_from_columns, &stmt.constraints)?;
+
+    let has_pk = row_id_index.is_none()
+        || stmt.constraints.iter().any(|cons| match cons {
+            TableConstraint::Unique { is_primary, .. } => *is_primary,
+            _ => false,
+        });
```

Comment:
There seems no need to check table constraints any more. It has been resolved in `bind_sql_table_constraints`.